src/Components/Base/Button/index.tsx

```diff
@@ -0,0 +1,42 @@
+import { useState } from 'react';
+import StyledButton from './style';
+import type { ButtonProp } from './type';
+
+const Button = ({
+  children,
+  color,
+  width,
+  height,
+  borderRound,
+  isToggle = false,
+  onClick = () => {},
+  onActiveChange = () => {},
+}: ButtonProp) => {
+  const [isActive, setIsActive] = useState(false);
+
+  const handleClick = () => {
+    if (isToggle) {
+      setIsActive(!isActive);
+      onActiveChange(isActive);
+
+      return;
+    }
+
+    onClick();
+  };
```

Comment:
개인적인 의견으로... onActiveChange랑 onClick 각각 받아와서 분리할 필요가 있을까 싶긴 하네요...?

어차피 함수의 내용은 저희가 버튼 컴포넌트 사용하면서 매개변수로 넘겨주며 정의 할거고, 여기서 이름만 이 함수로 사용하는거니까 그냥 `onClick 으로 매개변수(함수)를 통합` 시켜도 좋을 것 같아요. 만약 토글 버튼으로 사용하게 되면 어차피 onClick() 매개변수(함수)는 비어있을 거고, 그냥 일반 버튼으로 사용하게 된다면 onActiveChange가 비어있을 거라서...

```suggestion
  const handleToggle = () => {
      setIsActive(!isActive);
      onClick(isActive);
  }
  const handleClick = () => {
    onClick();
  };
```

```jsx
<StyledButton
      onClick={ isToggle ? handleToggle : handleClick }
      ....
>
```

이렇게 분리해보는 건 어떨까요? 하나의 함수에서 flag를 사용해 분기하는 것보다는, 하나의 함수에서는 하나의 기능만 하는게 좋을 것 같아서요 ㅎㅎ
혹시 제가 잘못 이해한게 있다면 말씀해주십쇼....

Reply:
onClick으로 IsActive를 전달할 생각을 못하고 있었네요..🥲 제안 주신 것처럼 onClick에서 IsToggle을 기준으로 로직을 다르게 처리하는 게 불필요한 콜백 사용을 줄일 수 있어 좋은 것 같습니다 👍🏻👍🏻